mp4parse_capi/src/lib.rs

```diff
@@ -989,11 +996,19 @@ fn create_sample_table(track: &Track, track_offset_time: i64) -> Option<Vec<mp4p
 
         let iter = sort_table.iter();
         for i in 0 .. (iter.len() - 1) {
-            let current_index = sort_table[i];
-            let peek_index = sort_table[i + 1];
-            let next_start_composition_time = sample_table[peek_index].start_composition;
-            let sample = &mut sample_table[current_index];
-            sample.end_composition = next_start_composition_time;
+            let (current_index, peek_index) = match (sort_table.get(i), sort_table.get(i + 1)) {
+                (Some(&current_index), Some(&peek_index)) => (current_index, peek_index),
+                _ => return None,
+            };
+
+            let next_start_composition_time = match sample_table.get(peek_index) {
```

Comment:
Do we need to use the fallible getters for peek_index and current_index?  Since they're extracted from sort_table which contains only valid indices into sample_table, it's impossible for these values to be out of range, unless I'm missing something.